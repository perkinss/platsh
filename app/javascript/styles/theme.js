import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
