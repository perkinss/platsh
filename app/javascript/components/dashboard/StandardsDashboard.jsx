import React from 'react'
import {Typography, Table, TableCell, TableRow, TableFooter, TablePagination, TableSortLabel, TableBody, TableHead,
    Icon, Chip, Avatar, LinearProgress, Grow, Tooltip, Zoom} from '@material-ui/core'
import Star from '@material-ui/icons/StarBorderRounded'
import { makeStyles } from "@material-ui/styles/index";

const useStyles = makeStyles(theme => ({
    progressBar: {
        minHeight: '28px',
        backgroundColor: theme.palette.grey[800],
        opacity: 1,
    },
}))

const standardBarStyle = makeStyles(theme => ({
    bar: {
        backgroundColor: 'rgba(0, 227, 150, 0.85)',
        borderRadius: '2px'
    }
}))

const topBarStyle = makeStyles(theme => ({
    bar: {
        backgroundColor: theme.palette.primary.light,
        borderRadius: '2px'
    }
}))

const tooltipStyles = makeStyles(theme => ({
    tooltip: {
        fontSize: '18px',
        marginLeft: '-120px',
        width: '60px'
    }
}))


export default function StandardsDashboard(props) {
    const { show, details, classes, theRef } = props
    const progressClasses = useStyles();
    const topBarClasses = topBarStyle();
    const tooltipClasses = tooltipStyles();
    const standardBarClasses = standardBarStyle()

    // TODO: remove the 'details.details' check when we create a competencies dashboard
    if (!show || !details.details) {
        return null
    }
    const classAverage = details.details.reduce((sum, detail) => {
        // TODO: format this calculation
        return sum + detail.average
    }, 0)/details.details.length

    return (
        <React.Fragment>
            <Typography paragraph variant={'h6'} ref={theRef}>{details.name||details.title}</Typography>
            <div >
            <Table size={'small'} autoFocus>
                <TableHead>
                    <TableRow>
                        <TableCell>Name</TableCell>
                        <TableCell colSpan={3}>
                            Number of tasks in level
                        </TableCell>
                        <TableCell width={'50%'}>
                            Class Average
                        </TableCell>
                    </TableRow>
                    <TableRow>
                        <TableCell/>
                        <TableCell padding={'none'} align={'center'}>
                            <Chip
                            avatar={
                                <Avatar>
                                    <Star/>
                                </Avatar>
                            }
                            label={
                                <React.Fragment>
                                    <Typography variant={'h6'}>L</Typography>
                                </React.Fragment>
                            }
                        />
                        </TableCell>
                        <TableCell padding={'none'} align={'center'}>
                        <Chip
                            avatar={
                                <Avatar>
                                    <Star/>
                                </Avatar>
                            }
                            label={
                                <React.Fragment>
                                    <Typography variant={'h6'}>M</Typography>
                                </React.Fragment>
                            }
                            color={'secondary'}
                            style={{marginLeft: '3px'}}
                        />
                        </TableCell>
                        <TableCell  padding={'none'} align={'center'}>
                        <Chip
                            avatar={
                                <Avatar>
                                    <Star/>
                                </Avatar>
                            }
                            label={
                                <React.Fragment>
                                    <Typography variant={'h6'}>H</Typography>
                                </React.Fragment>
                            }
                            color={'primary'}
                            style={{marginLeft: '3px'}}
                        />
                        </TableCell>
                        <TableCell>
                            <Tooltip classes={tooltipClasses} title={classAverage.toFixed(0) + '%'} placement="right-start" TransitionComponent={Zoom}>
                            <LinearProgress
                                classes={topBarClasses}
                                className={progressClasses.progressBar}
                                variant='determinate' value={classAverage}>
                                </LinearProgress>
                            </Tooltip>

                        </TableCell>
                    </TableRow>
                </TableHead>
                <TableBody >
                    {details.details.map((detail) => {

                        return (
                            <TableRow key={`row-${detail.description}`}>
                                <TableCell  >
                                    {detail.description}
                                </TableCell>
                                <TableCell  padding={'none'} align={'center'}>
                                    <Chip
                                        label={<Typography >{detail.l_count}</Typography>}
                                        className={classes.chip}
                                    />
                                </TableCell>
                                <TableCell padding={'none'} align={'center'} >
                                    <Chip
                                        label={<Typography> {detail.m_count}</Typography>}
                                        className={classes.chip}
                                        color={'secondary'}
                                    />

                                </TableCell>
                                <TableCell  padding={'none'} align={'center'}>
                                    <Chip
                                        label={<Typography> {detail.h_count}</Typography>}
                                        className={classes.chip}
                                        color={'primary'}
                                    />
                                </TableCell>
                                <TableCell with={'50%'}>
                                    <Grow
                                        in={show}
                                        style={{ transformOrigin: '0 0 0' }}
                                        timeout={1000}
                                    >
                                        <Tooltip classes={tooltipClasses} title={detail.average.toFixed(0) + '%'} placement="right-start" TransitionComponent={Zoom}>
                                            <LinearProgress
                                                color={'primary'}
                                                classes={standardBarClasses}
                                                className={progressClasses.progressBar}
                                                variant='determinate'
                                                value={detail.average}/>
                                        </Tooltip>
                                    </Grow>
                                </TableCell>
                            </TableRow>
                        )
                    })}

                </TableBody>
            </Table>
            </div>
        </React.Fragment>
    )

}