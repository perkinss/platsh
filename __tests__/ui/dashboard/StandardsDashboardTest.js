import React from 'react'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import StandardsDashboard from "../../../app/javascript/components/dashboard/StandardsDashboard";
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'

Enzyme.configure({ adapter: new Adapter() });

describe('<StandardsDashboard />', () => {
    let render;
    let classes = {
        content: {}
    }

    let section = {
        name: "New at the Zoo...",
        content_stats: [
            {name: "operations",
                id: 1,
                percent_included: "71",
                details: [
                    {
                        average: 80,
                        description: "write and evaluate Powers",
                        h_count: 0,
                        id: 8,
                        l_count: 0,
                        m_count: 2
                    }
                    ]
            },
            {name: "exponents", id: 2, percent_included: "0",details: [
                    {
                        average: 80,
                        description: "write and evaluate Powers",
                        h_count: 0,
                        id: 8,
                        l_count: 0,
                        m_count: 2
                    }]
                    },
        ]
    }
    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                    <StandardsDashboard classes={classes} section={section} />
            </ThemeProvider>
                );
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should have the details', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <StandardsDashboard classes={classes} show={true} details={section.content_stats[0]} />
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toContain("operations")
        expect(result).not.toContain("exponents")
    });

    it('should remain hidden if not showing', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <StandardsDashboard classes={classes} show={false} details={section.content_stats[0]} />
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toBeNull()
    })

    it('should remain hidden if no details', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <StandardsDashboard classes={classes} show={true} details={''}/>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toBeNull()
    })

})