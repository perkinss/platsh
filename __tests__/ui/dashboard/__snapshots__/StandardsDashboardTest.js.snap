// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StandardsDashboard /> should match the snapshot 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
